packages/smithy-client/src/ser-utils.ts

```diff
@@ -20,3 +20,11 @@ export const serializeFloat = (value: number): string | number => {
       return value;
   }
 };
+
+/**
+ * @param date - to be serialized.
+ * @returns https://smithy.io/1.0/spec/core/protocol-traits.html#timestampformat-trait date-time format.
+ */
+export const serializeDateTime = (date: Date): string => {
+  return date.toISOString();
```

Comment:
this is a bit redundant at a glance, but leaves the option to modify truncation or formatting later without a huge codegen diff